Kernel/Heap/kmalloc.cpp

```diff
@@ -65,19 +66,32 @@ class KmallocSlabBlock {
         }
     }
 
-    void* allocate()
+    void* allocate([[maybe_unused]] size_t requested_size)
     {
         VERIFY(m_freelist);
         ++m_allocated_slabs;
-        return exchange(m_freelist, m_freelist->next);
+#ifdef HAS_ADDRESS_SANITIZER
+        AddressSanitizer::fill_shadow((size_t)m_freelist, sizeof(FreelistEntry::next), Kernel::AddressSanitizer::ShadowType::Unpoisoned8Bytes);
+#endif
+        auto* ptr = exchange(m_freelist, m_freelist->next);
+#ifdef HAS_ADDRESS_SANITIZER
+        AddressSanitizer::mark_region((size_t)ptr, requested_size, m_slab_size, AddressSanitizer::ShadowType::Malloc);
```

Comment:
These should probably be FlatPtr, not size_t?

Reply:
Sure, changed all to FlatPtr. I initially used size_t everywhere inside AddressSanitizer.cpp, since I thought the C++ `using` type wouldn't work in the extern "C" definitions for some reason, but it does.